### ISSUE TITLE
Upgrade redo package to 2.0.1

### ISSUE THREAD

**Issue by samueldg** [enigmampc/catalyst#494]
Yesterday, the mozilla/releng folks released version [2.0.1](https://pypi.org/project/redo/2.0.1/) of the `redo` package, which Catalyst uses in `get_exchange_data`. That function is one of the "hottest" parts of the code, and is executed very frequently.

Version 2.0.1 of `redo` includes a significant performance fix I made specifically based off my profiling of a Catalyst run.

There are more details in my PR: https://github.com/mozilla-releng/redo/pull/51, 
the TL;DR being that upgrading this library would shave off around 10% of the time it takes to run a long backtest.

**Comment by lenak25** [comment]
@samueldg , thanks for this improvement! 
We have updated the requirements files, so the next catalyst release will include the upgraded version of redo.

**Comment by lenak25** [comment]
Included in released 0.5.21.
